### PR TITLE
Add instance status column. Implements #572

### PR DESCRIPTION
**Pull request by dmfs**
This commit adds an instance status column which indicates which instance of a task is the next one to complete. This allows to filter out the instances which come after the next one.

### CODE REVIEW

File: opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/instancedata/Stated.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import android.content.ContentValues;
+
+import org.dmfs.jems.single.Single;
+import org.dmfs.jems.single.decorators.DelegatingSingle;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * A {@link Single} of date and time {@link ContentValues} of an instance.
```
Comment (lemonboston) [inline opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/instancedata/Stated.java]: Not updated, copy-pasted javadoc here.

File: opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskDatabaseHelper.java [hunk 2]
```diff
@@ -237,7 +237,8 @@
                     + TaskContract.Instances.INSTANCE_START_SORTING + " INTEGER, "
                     + TaskContract.Instances.INSTANCE_DUE_SORTING + " INTEGER, "
                     + TaskContract.Instances.INSTANCE_DURATION + " INTEGER, "
-                    + TaskContract.Instances.INSTANCE_ORIGINAL_TIME + " INTEGER DEFAULT 0);";
+                    + TaskContract.Instances.INSTANCE_ORIGINAL_TIME + " INTEGER DEFAULT 0,"
```
[the remark this answers is not in the document]
Comment (dmfs) [reply opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskDatabaseHelper.java]: no, technically it's not required

File: opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/instancedata/Stated.java (hunk 1 above)
Comment (dmfs) [reply opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/instancedata/Stated.java]: fixed

File: opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java [hunk 3]
```diff
@@ -239,7 +244,41 @@ public void testInsertTaskWithStartAndDue()
...
-                        new EqArg(Tasks.TZ, start.isAllDay() ? "UTC" : start.getTimeZone().getID())
+                        new EqArg(Tasks.TZ, start.isAllDay() ? "UTC" : start.getTimeZone().getID()),
+                        new EqArg(Instances.INSTANCE_STATUS, Instances.INSTANCE_STATUS_NEXT)
+                ))
+        ));
+    }
+
+
+    /**
+     * Create task with start and due, check datetime values including generated duration.
```
[the remark this answers is not in the document]
Comment (dmfs) [reply opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java]: fixed

**Comment by dmfs** [comment]
@lemonboston please re-review.

**Pull request merged**